doc/library/gpuarray/dnn.txt

```diff
@@ -132,30 +132,70 @@ To get an error if Theano can not use cuDNN, use this Theano flag:
     cudnn.h must be readable by everybody.
 
 
-Functions
-=========
+* :ref:`Convolution <libdoc_gpuarray_dnn_convolution>`
+* :ref:`Pooling <libdoc_gpuarray_dnn_pooling>`
+* :ref:`Batch Normalization <libdoc_gpuarray_dnn_bn>`
+* :ref:`RNN <libdoc_gpuarray_dnn_rnn>`
+* :ref:`Softmax <libdoc_gpuarray_dnn_softmax>`
+* :ref:`Internal Ops <libdoc_gpuarray_dnn_internal_ops>`
+
+dnn_present, dnn_available
+
+.. _libdoc_gpuarray_dnn_convolution:
+
+Convolution
+===========
 
 .. automodule:: theano.gpuarray.dnn
    :noindex:
-   :members: dnn_conv, dnn_pool
+   :members: dnn_conv, dnn_conv3d, dnn_gradweight, dnn_gradweight3d, dnn_gradinput, dnn_gradinput3d
+
+.. _libdoc_gpuarray_dnn_pooling:
 
-Convolution Ops
-===============
+Pooling
+=======
 
 .. automodule:: theano.gpuarray.dnn
    :noindex:
-   :members: GpuDnnConvDesc, GpuDnnConv, GpuDnnConvGradW, GpuDnnConvGradI
+   :members: dnn_pool
 
-Pooling Ops
-===========
+.. _libdoc_gpuarray_dnn_bn:
+
+Batch Normalization
+===================
 
 .. automodule:: theano.gpuarray.dnn
    :noindex:
-   :members: GpuDnnPoolDesc, GpuDnnPool, GpuDnnPoolGrad
+   :members: dnn_batch_normalization_train, dnn_batch_normalization_test
 
-Softmax Ops
-===========
+.. _libdoc_gpuarray_dnn_rnn:
+
+RNN
+===
+
+New back-end only! Without dropout support.
```

Comment:
I would say that the warning about new backend only is unnecessary since this is in gpuarray/dnn.txt.